### DISCUSSION TITLE
Issue with Nuxt 3 Layers and Nuxt i18n

### DISCUSSION THREAD

**Discussion by ionut-sg** [nuxt-modules/i18n#2388] [Q&A]
### Issue Description
**Package Name/Repository:** Nuxt 3 with Layers

**Nuxt i18n Version:** "@nuxtjs/i18n": "v8.0.0-rc.4"

### Issue Details
I am currently working with Nuxt 3 Layers for my project. When I extend Layers from local repositories, everything works perfect. However, when I try to extend Layers from a GitHub repository link, I encounter an issue with the Nuxt i18n package. The specific error message I'm getting is:

`ERROR  No lockfile found from /***/node_modules/base-app`

_**It's important to note that I have the latest version of "@nuxtjs/i18n" (v8.0.0-rc.4) installed in all Layers.**_

### Steps to Reproduce
To reproduce this issue, follow these steps:

- Create a Nuxt 3 project as a base Layer.

- Use Nuxt i18n in the base Layer.

- Push the project to a GitHub repository.

- Create a Nuxt 3 project which extends the base Layer from the GitHub repository link.

- Install the dependencies.

_**The base-app layer is added as a dependency in package.json as follows:**_
`"dependencies": {
    "base-app": "git+ssh://github_pat_***@github.com:***/base-app.git"
  }`

### Expected Behavior
I expected the project to run without errors when extending Layers from a GitHub repository while using the latest version of Nuxt i18n.

### Actual Behavior
Instead, I encountered the "No lockfile found" error as described above.

### Additional Notes
I'd like to highlight that the described issue appears to be introduced in the latest versions of the library. To provide some context, I experimented with an alpha version of the library, and I couldn't reproduce the issue. This suggests that there might be a regression or compatibility issue in the most recent releases of the library, specifically affecting projects using Nuxt 3 Layers in conjunction with Nuxt i18n.

cc/ @kazupon 
I would appreciate any assistance or insights from the community or individuals experienced with Nuxt 3 Layers and Nuxt i18n.

**Comment by skoenfaelt** [comment]
I can confirm. Okay i thought the issue came with `yarn npm publish`
https://github.com/npm/cli/issues/6738

But how to solve that? its a huge blocker

Noteworthy, if you put the yarn.lock from your layer into the node_modules/your_layer of the app, you receive the following error:

Cannot resolve vue-i18n on yarn! please install it on 'node_modules'

But in package.json of the layer:

```
 "dependencies": {
    ...
    "@nuxtjs/i18n": "^8.0.0-rc.5",
    "vue-i18n": "^9.5.0",
    ...
  },
  ```

**Comment by skoenfaelt** [comment]
Node Module Installation

So I cant give further information, I will see how I can get beta.4 working for production, while this issue on newest versions persist.

```
"@nuxtjs/i18n": "8.0.0-beta.4", // working


"@nuxtjs/i18n": "8.0.0-beta.5", // and above is not working => [90mat [39masync getPackageManagerType ([36m[...]/node_modules/@nuxtjs/i18n/dist/module.mjs:87:26[39m)
```

Following packages have changed 
```
// added
+ @nuxtjs/i18n@npm:8.0.0-beta.5, @intlify/bundle-utils@npm:4.0.0, @intlify/shared@npm:9.3.0-beta.10, @intlify/unplugin-vue-i18n@npm:0.8.2, vue-i18n-routing@npm:0.10.3

// removed
- @intlify/shared@npm:9.3.0-beta.6, @intlify/unplugin-vue-i18n@npm:0.7.3, @nuxtjs/i18n@npm:8.0.0-beta.4, knitwork@npm:0.1.3, mlly@npm:0.5.17, pathe@npm:0.3.9, ufo@npm:0.8.6, unplugin@npm:0.8.1, unplugin@npm:0.9.6, vue-i18n-routing@npm:0.9.2, webpack-virtual-modules@npm:0.4.6

```

![image](https://github.com/nuxt-modules/i18n/assets/53934248/49565a5f-64ae-4764-ad87-6cda8aa4e7d4)

```
# This file contains the result of Yarn building a package (@q-soft/doc-tool@npm:1.0.0-beta.63::__archiveUrl=http%3A%2F%2Fefqstooling%2Fapi%2Fv4%2Fprojects%2F272%2Fpackages%2Fnpm%2F%40q-soft%2Fdoc-tool%2F-%2F%40q-soft%2Fdoc-tool-1.0.0-beta.63.tgz)
# Script name: postinstall


[41m[30m ERROR [39m[49m No lockfile found from C:\Users\christophers\repos\qs-docu-tool\src\docs\node_modules\@q-soft\doc-tool

  [90mat [39mresolveLockfile ([36m/C:/Users/christophers/repos/qs-docu-tool/src/docs/node_modules/pkg-types/dist/index.mjs:122:9[39m)
  [90mat [39masync getPackageManagerType ([36m/C:/Users/christophers/repos/qs-docu-tool/src/docs/node_modules/@nuxtjs/i18n/dist/module.mjs:76:26[39m)
  [90mat [39masync setupAlias ([36m/C:/Users/christophers/repos/qs-docu-tool/src/docs/node_modules/@nuxtjs/i18n/dist/module.mjs:110:18[39m)
  [90mat [39masync setup ([36m/C:/Users/christophers/repos/qs-docu-tool/src/docs/node_modules/@nuxtjs/i18n/dist/module.mjs:855:5[39m)
  [90mat [39masync normalizedModule ([36m/C:/Users/christophers/repos/qs-docu-tool/src/docs/node_modules/@nuxt/kit/dist/index.mjs:2141:17[39m)
  [90mat [39masync installModule ([36m/C:/Users/christophers/repos/qs-docu-tool/src/docs/node_modules/@nuxt/kit/dist/index.mjs:2439:95[39m)
  [90mat [39masync initNuxt ([36m/C:/Users/christophers/repos/qs-docu-tool/src/docs/node_modules/nuxt/dist/index.mjs:3653:7[39m)
  [90mat [39masync loadNuxt ([36m/C:/Users/christophers/repos/qs-docu-tool/src/docs/node_modules/nuxt/dist/index.mjs:3736:5[39m)
  [90mat [39masync loadNuxt ([36m/C:/Users/christophers/repos/qs-docu-tool/src/docs/node_modules/@nuxt/kit/dist/index.mjs:2584:19[39m)
  [90mat [39masync Object.run ([36m/C:/Users/christophers/AppData/Roaming/npm/node_modules/nuxi/dist/chunks/prepare.mjs:58:18[39m) 



[41m[30m ERROR [39m[49m No lockfile found from C:\Users\christophers\repos\qs-docu-tool\src\docs\node_modules\@q-soft\doc-tool
```

**Comment by skoenfaelt** [comment]
Issue according this Problem https://github.com/nuxt-modules/i18n/issues/2491
